diff --git a/kommon.h b/kommon.h
index b913549..fb8c3f5 100644
--- a/kommon.h
+++ b/kommon.h
@@ -19,4 +19,4 @@
 #define kom_roundup32(x) (--(x), (x)|=(x)>>1, (x)|=(x)>>2, (x)|=(x)>>4, (x)|=(x)>>8, (x)|=(x)>>16, ++(x))
-#define kom_roundup64(x) (--(x), (x)|=(x)>>1, (x)|=(x)>>2, (x)|=(x)>>4, (x)|=(x)>>8, (x)|=(x)>>16, (x)|=(x)>>32, ++(x))
+#define kom_roundup64(x) (--(x), (x)|=(x)>>1, (x)|=(x)>>2, (x)|=(x)>>4, (x)|=(x)>>8, (x)|=(x)>>16, (x)|=(uint64_t)(x)>>32, ++(x))
-
+
 #define kom_reverse(type, n, a) do { \
